lightning/src/ln/msgs.rs

```diff
@@ -1161,6 +1163,9 @@ impl Writeable for Init {
 	fn write<W: Writer>(&self, w: &mut W) -> Result<(), io::Error> {
 		// global_features gets the bottom 13 bits of our features, and local_features gets all of
 		// our relevant feature bits. This keeps us compatible with old nodes.
+		encode_tlv_stream!(w, {
```

Comment:
Read and write have to match exactly - note that you're writing the TLV stream here before the features, but reading features first later. Please add tests for reading as well as writing (like we do for some other messages).

Reply:
I am also getting this warning because of the above code
```
warning: unused `Result` that must be used
    --> lightning/src/ln/msgs.rs:1167:3
     |
1167 |         self.features.write(w);
     |         ^^^^^^^^^^^^^^^^^^^^^^^
     |
     = note: `#[warn(unused_must_use)]` on by default
```

Reply:
You can just fix the warnings, instead of asking about them :p.

Reply:
Well i tried removing this line as it is unused. But doing so shortens the hex codes of the tests, so i thought removing this line might remove some `features` details that were supposed to be written

Reply:
I'm not if I should remove this or not

Reply:
The warning suggests that you have to do *something* with the `Result` that is returned, not just remove the line. The `Result` may indicate a failure, which should be handled, just like other failures in the function.

Reply:
You can read more about errors and results in Rust by reading the Rust book - https://doc.rust-lang.org/book/ch09-02-recoverable-errors-with-result.html